frontend/src/components/eventPage/EventPage.jsx

```diff
@@ -1,48 +1,61 @@
-import React, { useState, useEffect } from "react";
-import Header from "../header/Header";
-import DisplayEventInfo from "./DisplayEventInfo";
-import DisplayEventComments from "./DisplayEventComments";
+import { useState, useEffect } from 'react'
+import Header from '../header/Header'
+import DisplayEventInfo from './DisplayEventInfo'
+import DisplayEventComments from './DisplayEventComments'
+import LoadingPage from '../loadingPage/LoadingPage'
 
 const EventPage = () => {
-  const current_link = window.location.pathname;
-  const current_eventId = current_link.split("/")[2];
-  const [event, setEvent] = useState();
-  const [comments, setComments] = useState();
-  let [changeInEvent, setchangeInEvent] = useState(true);
+    const current_link = window.location.pathname
+    const current_eventId = current_link.split('/')[2]
+    const [event, setEvent] = useState()
+    const [comments, setComments] = useState()
+    let [changeInEvent, setchangeInEvent] = useState(true)
+    const [isLoading, setIsLoading] = useState(true)
 
-  useEffect(() => {
-    if (changeInEvent) {
-      fetchEvent();
-      fetchComments();
-    }
-    setchangeInEvent(false);
-  }, [event]);
+    useEffect(() => {
+        if (changeInEvent) {
+            setIsLoading(true)
+            fetchEvent()
+            fetchComments()
+        }
+        setIsLoading(false)
```

Comment:
`isLoading` should be based on the result of your data fetch. `fetch` is a promise, which can be async/awaited. Read this as an example, https://devtrium.com/posts/async-functions-useeffect

Reply:
Read this to understand async/await better, https://javascript.info/async-await

Reply:
To Do:
Fix isLoading to be based on the result of the data fetch